test/unit/services/s3.test.js

```diff
@@ -80,3 +86,138 @@ describe('listObjects', () => {
     ]);
   });
 });
+
+// TODO: test listRawFileNames. this function calls listObjects which makes a call to aws api
+// might need to refactor listRawFileNames to allow for mocking a return value for listObjects
+
+describe('getObject', () => {
+  test('Returns a buffer', async () => {
+
+  // set up mocks
+  const sampleS3Object = {
+    Key: 'staging/IMG_20161216_142249.jpg',
+    LastModified: '2022-07-27T03:17:56.000Z',
+    ETag: '"4727374fff1eb029687319184bc23a10"',
+    ChecksumAlgorithm: [],
+    Size: 1761676,
+    StorageClass: 'STANDARD',
+    Body: 'sample buffer',
+  }
+  const testGetObject = jest.fn().mockReturnValue({
+    promise: () => Promise.resolve(sampleS3Object),
+  });
+  const s3 = {
+    getObject: testGetObject,
+  };
+  const { Key: sampleKey } = sampleS3Object;
+
+  // test
+  const getObjectResult = await getObject(sampleKey, s3);
+
+  // asssert
+  expect(getObjectResult).toEqual(sampleS3Object.Body)
+  });
+})
+
+describe('moveObject', () => {
+  test('Calls s3.copyObject', async () => {
+
+    // set up mocks
+    const deleteObject = jest.fn().mockReturnValue({
+      promise: () => Promise.resolve(),
+    });
+    const copyObject = jest.fn().mockReturnValue({
+      promise: () => Promise.resolve(),
+    });
+
+    const s3 = {
+      copyObject,
+      deleteObject,
+    }
+
+    //test
+    const result = await moveObject('test-origin-folder/test-s3-object.png', 'test-origin-folder/', 'test-destination-folder/', s3)
+    
+    // assert
+    expect(copyObject).toHaveBeenCalled();
+  });
+
+  test('Calls s3.deleteObject', async () => {
```

Comment:
This test seems to have an identical setup to the previous test. You can merge them by simply adding `expect(deleteObject).toHaveBeenCalled();` to the assertions 👍

Reply:
Oops. I think I wanted to separate tests for either call in case there's an error with just one of them, but I guess that's the point of each individual assertion lol

Reply:
Yep! When a test requires a different setup (e.g. different input data), then that warrants a separate test